pkg/resources/custom_diffs.go

```diff
@@ -98,6 +98,22 @@ func ComputedIfAnyAttributeChanged(key string, changedAttributeKeys ...string) s
 	})
 }
 
+// TODO(SNOW-1629468): Adjust the function to make it more flexible
+func ComputedIfAnyAttributeChangedWithSuppressDiff(key string, suppressDiffFunc schema.SchemaDiffSuppressFunc, changedAttributeKeys ...string) schema.CustomizeDiffFunc {
+	return customdiff.ComputedIf(key, func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) bool {
+		for _, changedKey := range changedAttributeKeys {
+			if diff.HasChange(changedKey) {
+				oldValue, newValue := diff.GetChange(changedKey)
+				if !suppressDiffFunc(key, oldValue.(string), newValue.(string), nil) {
+					log.Printf("[DEBUG] ComputedIfAnyAttributeChangedWithSuppressDiff: changed key: %s", changedKey)
+					return true
```

Comment:
Nit: is there a reason we return early here? Why don't we log all diffs, like in ComputedIfAnyAttributeChanged?

Reply:
I mean, I just returned early, because other changes don't matter. If one thing changed, we want to recompute anyway, so there's no point of checking other keys. I guess this should be sorted out in the upcoming change where we want to change this function (or/and merge with the other one).